src/dawn/CodeGen/CodeGen.cpp

```diff
@@ -8,6 +8,20 @@ size_t CodeGen::getVerticalTmpHaloSize(Stencil const& stencil) {
   return (tmpInterval != nullptr) ? std::max(tmpInterval->overEnd(), tmpInterval->belowBegin()) : 0;
 }
 
+size_t CodeGen::getVerticalTmpHaloSizeForMultipleStencils(
+    const std::vector<std::shared_ptr<Stencil>>& stencils) const {
+  std::shared_ptr<Interval> fullIntervals = stencils[0]->getEnclosingIntervalTemporaries();
+  for(auto stencil : stencils) {
+    auto tmpInterval = stencil->getEnclosingIntervalTemporaries();
+    if(tmpInterval != nullptr) {
+      fullIntervals->merge((*tmpInterval));
```

Comment:
If the first stenci has no temporaries, fullIntervals will be nullptr and this will segfault (on the second iteration)